public/app/plugins/panel/timeseries/plugins/ExemplarMarker.tsx

```diff
@@ -202,20 +202,27 @@ export const ExemplarMarker = ({
     .getSeries()
     .find((s) => s.props.dataFrameFieldIndex?.frameIndex === dataFrameFieldIndex.frameIndex)?.props.lineColor;
 
+  const onExemplarClick = () => {
+    setClickedExemplarFieldIndex(dataFrameFieldIndex);
+    lockExemplarModal();
+  };
+
   return (
     <>
-      {/* TODO: fix keyboard a11y */}
-      {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */}
       <div
         ref={setMarkerElement}
-        onClick={() => {
-          setClickedExemplarFieldIndex(dataFrameFieldIndex);
-          lockExemplarModal();
+        onClick={onExemplarClick}
+        onKeyDown={(e: React.KeyboardEvent) => {
+          if (e.key === 'Enter') {
+            onExemplarClick();
+          }
```

Comment:
On key down I noticed while testing that it is possible to open an exemplar with the ender button while having one open on hover via the mouse.

This is a pretty minor edge case so if it's not easy to fix we can probably just create a follow-up issue to improve a11y of exemplars overall (i.e. tooltip popup x button isn't tabbable - though this is something we will need to address in our tooltip rewrite)


https://github.com/grafana/grafana/assets/22381771/abdff341-2d9a-447b-84c4-4f0bacbeb3ee

Reply:
As I discovered each ExemplarMarker component creates its own pop-up. One ExemplarMarker instance can't just say to another ExemplarMarker to close its popup 😁 That would be cool anyway, horizontal communication between components sounds futuristic 😎

I think we need to lift up this Portal/pop-up element somewhere behind ExemplarMarker to be able to manipulate only one instance. But yeah, it's totally different task.

Nice catch btw 👍🏻